### PR TITLE
Update js challenges

### PR DESCRIPTION
**Pull request by mikelpmc**

### CODE REVIEW

File: staff/mikelpmc/js/challenges/scripts/cube.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+ * @param {number|number[]} numbers
+ *
+ * @return {string} - Comma separated string with the cubes of the given numbers
+ */
 function cube(numbers) {
     return numbers
         .toString()
         .split(',')
         .map(n => parseInt(Math.pow(n, 3)))
         .join(', ');
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/scripts/cube.js]: WARN! cube should NOT return a string, but an array (if input is an array of number) or a number (if input is a number). on the other hand, it should NOT accept non-numeric values in input, neither inside the array.

File: staff/mikelpmc/js/challenges/scripts/map.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @param {Function} handler - Function to transform data in the array
+ *
+ * @returns {Array} - New array with each item transformed with the handler function
+ */
+function map(arr) {
+    if (!(arr instanceof Array)) throw Error('Input arr should be an array');
+
+    var newArr = [];
+
+    return function(handler) {
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/scripts/map.js]: ERROR -> why is map returning a function!? 

RTFM -> search in google "mdn js Array.prototype.map".

File: staff/mikelpmc/js/challenges/scripts/unit-testing-tool-3.0.0.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+'use strict';
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/scripts/unit-testing-tool-3.0.0.js]: MAY you update it to the new version 3.1.0.

**Comment by manuelbarzi** [comment]
REVIEW your pr, SEE comments, CORRECT your code, and APPLY changes and PUSH them.

File: staff/mikelpmc/hangman-app/scripts/prueba.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+var word = 'hello';
```
Comment (agandia9) [inline staff/mikelpmc/hangman-app/scripts/prueba.js]: ¿?

File: staff/mikelpmc/js/challenges/scripts/Array.prototype.filter.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+if (!(Array.prototype.filter instanceof Function)) {
```
Comment (agandia9) [inline staff/mikelpmc/js/challenges/scripts/Array.prototype.filter.js:3]: use strict.

File: staff/mikelpmc/js/challenges/scripts/Array.prototype.wrap.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+// - wrap
```
Comment (agandia9) [inline staff/mikelpmc/js/challenges/scripts/Array.prototype.wrap.js:3]: use strict.

**Pull request merged**